pkg/agents/steprunner/step_runner.go

```diff
@@ -140,29 +154,43 @@ func (a *agent) Start(ctx context.Context) error {
 
 	// Start the "job runner"
 	go func() {
+		timer := time.NewTimer(a.pollingInterval)
 		for {
-			if err := a.run(ctx, gsRunner); err != nil {
-				if !errors.Is(err, gorm.ErrRecordNotFound) {
-					slog.Error("failed step runner iteration", "err", err)
+			a.run(ctx, gsRunner)
+			select {
+			case <-ctx.Done():
+				// Ensure the timer channel is drained
+				if !timer.Stop() {
+					select {
+					case <-timer.C:
+					default:
+					}
 				}
+				return
+			case <-timer.C:
+			case <-a.trigger.Triggered():
+			}
```

Comment:
Since this select statement is no longer predicated by an error from `a.run` AND the trigger implementation _doesn't_ block on send to the unbuffered `syncNow` channel, is it possible that we'll "miss" notifications when there's more than one concurrent run and end up waiting until the timer expires? 

Am I overlooking some detail that renders this case moot?

Reply:
The original implementation would completely process the run step before getting back to this logic. The idea was that if we completely processed a run step, then we almost assuredly did so for longer than a second and we should immediately check again.

Now, we kick off the processing in a separate goroutine and come back to this logic. Therefore, if we miss a trigger and wait a second, then the next run step is still processed much sooner than the previous implementation.

Let me know if you disagree with this approach, or if you have another idea.